src/model/mixin/itemStyle.ts

```diff
@@ -54,15 +64,24 @@ class ItemStyleMixin {
         includes?: readonly (keyof ItemStyleOption)[]
     ): ItemStyleProps {
         const style = getItemStyle(this, excludes, includes);
-        const lineDash = this.getBorderLineDash();
-        lineDash && (style.lineDash = lineDash);
+        style.lineDash = this.getBorderLineDash(style.lineWidth);
         return style;
     }
 
-    getBorderLineDash(this: Model): number[] {
+    getBorderLineDash(this: Model, lineWidth?: number): number[] {
         const lineType = this.get('borderType');
-        return (lineType === 'solid' || lineType == null) ? null
-            : (lineType === 'dashed' ? [5, 5] : [1, 1]);
+        if (lineType == null || lineType === 'solid') {
+            return [];
+        }
+
+        lineWidth = lineWidth || 0;
+
+        let dashArray = this.get('borderDashArray');
+        // compatible with single number
+        if (dashArray != null && !isNaN(dashArray)) {
+            dashArray = [+dashArray];
+        }
+        return dashArray || (lineType === 'dashed' ? [5, 5] : [1, 1]);
     }
```

Comment:
I'm not sure, but since these two options can only have one work, can we merge them into one option?

Like `borderType` can be an array to represent `borderDashArray`:

```ts
borderType: 'solid' | 'dashed' | 'dotted' | number[]
```

Similar question on `lineType` and `textBorderType`

Reply:
That should be feasible and can reduce the options. I'll tweak it later.

Reply:
@pissang There is an issue puzzling me. We added a semantic value `bolder` in #13013, but I found this special value will affect the calculation of default dash/dot size and return an illegal value `NaN`. Perhaps, we should change the logic about either `LineView` or `model/mixin/lineStyle`?

https://github.com/apache/incubator-echarts/blob/next/src/chart/line/LineView.ts#L553-L557
https://github.com/apache/incubator-echarts/blob/next/src/model/mixin/lineStyle.ts#L53
https://github.com/apache/incubator-echarts/blob/next/src/model/mixin/lineStyle.ts#L62-L63

Reply:
The second issue:
if we make `borderDashArray/dashArray(line)` to be a value of `borderType/type(line)`, 
the `ITEM_STYLE_KEY_MAP` in `model/mixin/itemStyle.ts` will miss the property `borderDash` and 
the `LINE_STYLE_KEY_MAP` in `model/mixin/lineStyle.ts` will miss the property `lineDash`.

This two missing properties can't be get by `getItemVisual`, so dash style will be not working in some cases like `lines`, 
refer to [chart/helper/Line.ts#L206](https://github.com/apache/incubator-echarts/blob/next/src/chart/helper/Line.ts#L206). 

It's different from [line series](https://github.com/apache/incubator-echarts/blob/next/src/chart/line/LineView.ts#L543) and [4.x](https://github.com/apache/incubator-echarts/blob/master/src/chart/helper/Line.js#L374) which are using `lineStyleModel.getLineStyle()`

So how should I do to make it work for those series dependent on `chart/helper/Line.js`?